File: test/fixtures/unions/android/src/main/java/com/example/unions/UnionsPlugin.java
package com.example.unions;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Unions")
public class UnionsPlugin extends Plugin {

    private String flavor = "sweet";
    private String origin = "north";

    @PluginMethod
    public void getTaste(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("flavor", flavor);
        ret.put("heat", "low");
        ret.put("origin", origin);
        ret.put("note", JSObject.NULL);
        call.resolve(ret);
    }

    @PluginMethod
    public void setFlavor(PluginCall call) {
        flavor = call.getString("flavor");
        call.resolve();
    }

    @PluginMethod
    public void getDirection(PluginCall call) {
        call.resolve(flavor);
    }

    @PluginMethod
    public void getRawDirection(PluginCall call) {
        call.resolve(origin);
    }

    @PluginMethod
    public void getNote(PluginCall call) {
        call.resolve();
    }
}

File: test/fixtures/unions/ios/Sources/UnionsPlugin.swift
import Foundation
import Capacitor

@objc(UnionsPlugin)
public class UnionsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "UnionsPlugin"
    public let jsName = "Unions"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getTaste", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "setFlavor", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getDirection", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getRawDirection", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getNote", returnType: CAPPluginReturnPromise)
    ]

    private var flavor = "sweet"
    private var origin = "north"

    @objc func getTaste(_ call: CAPPluginCall) {
        call.resolve([
            "flavor": flavor,
            "heat": "low",
            "origin": origin,
            "region": NSNull(),
            "note": NSNull()
        ])
    }

    @objc func setFlavor(_ call: CAPPluginCall) {
        flavor = call.getString("flavor") ?? "sweet"
        call.resolve()
    }

    @objc func getDirection(_ call: CAPPluginCall) {
        call.resolve(flavor)
    }

    @objc func getRawDirection(_ call: CAPPluginCall) {
        call.resolve(origin)
    }

    @objc func getNote(_ call: CAPPluginCall) {
        call.resolve()
    }
}
